apps/masterbots.ai/lib/helpers/ai-streams.ts

```
@@ -0,0 +1,97 @@
+  try {
+    while (true) {
+      const { done, value } = await reader.read()
+      if (done) break
+
+      buffer += decoder.decode(value, { stream: true })
+
+      // console.log('Buffer: [Before Changes] => ', buffer)
+
+      let match
+      let lastIndex = 0
+      while ((match = jsonRegex.exec(buffer)) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)